### ISSUE TITLE
Media type for SOAP response should be text/xml

### ISSUE THREAD

**Issue by candlerb** [zatosource/zato#462]
When a service is invoked on a SOAP channel, zato includes `Content-Type: application/xml` in the reply. However the required media type is `text/xml`.

See the [SOAP 1.1 spec](http://www.w3.org/TR/2000/NOTE-SOAP-20000508/#_Toc478383526)

> (6) HTTP applications MUST use the media type "text/xml" according to RFC 2376 [3] when including SOAP entity bodies in HTTP messages.

and the [WS-Basic Profile 1.0](http://www.ws-i.org/profiles/BasicProfile-1.0-2004-04-16.html#refinement16485352)

> (4.3.8) ... the Profile highlights those when the payload of the HTTP request is not the proper media type (i.e., "text/xml", as required by the SOAP/HTTP binding)

**Comment by dsuch** [comment]
The latter spec build on the former which in turn leans on RFC 2376 but I have absolutely no idea how according that RFC the media type should be ```text/xml``` instead of ```application/xml``` or either for that matter.

I understand it doesn't matter but I just don't follow the 'according to' point.

At any rate, would you like me to tell you how to make the default value be different or you're just signalling that it need be changed?

**Comment by candlerb** [comment]
You're right, RFC 2376 just defines both.

I think zato's behaviour should be changed, it seems to do the job if I change `zato-server/src/zato/server/spring_context.py`:

~~~
...
    def soap11_content_type(self):
        return 'text/xml' # 'application/xml'
...
~~~

I'd also like to point out that the current behaviour is inconsistent between outbound and inbound channels. On an unpatched system: if I create one service which invokes an outbound SOAP channel to talk to another SOAP channel, tcpdump shows that the outbound connection is using `text/xml; charset=utf-8`, but the response is `application/xml`

~~~
POST /testing/dummy HTTP/1.1
Host: localhost:11223
Content-Length: 210
Accept-Encoding: gzip, deflate
SOAPAction:
Accept: */*
User-Agent: python-requests/2.7.0 CPython/2.7.6 Linux/3.13.0-54-generic
Connection: keep-alive
X-Zato-Msg-TS: 2015-06-19T18:15:50.734497
X-Zato-Component: parallel/cfprov/cfprov/6412/Dummy-1638
X-Zato-CID: K06HR3K5EEYHCYFPG9S9JEY6Y4EK
Content-Type: text/xml; charset=utf-8

<?xml version="1.0" encoding="utf-8"?>
<s11:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:s11="http://schemas.xmlsoap.org/soap/envelope/">

  <s11:Body><bar/></s11:Body>
</s11:Envelope>
~~~

...

~~~
HTTP/1.1 200 OK
Server: Zato
Date: Fri, 19 Jun 2015 18:15:55 GMT
Connection: close
Transfer-Encoding: chunked
Content-Type: application/xml
X-Zato-CID: K04P80Y29ACKFB5QW71DP29F4WAP

BD
<?xml version='1.0' encoding='UTF-8'?><soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns="https://zato.io/ns/20130518"><soap:Body><foo/></soap:Body></soap:Envelope>
0
~~~

The service code does not set request or response headers:

~~~
from __future__ import absolute_import, division, print_function, unicode_literals
from zato.server.service import Service
import time

class DummyReceive(Service):
    def handle(self):
        self.logger.info("Request: %s" % self.request.raw_request)
        time.sleep(5)
        self.response.payload = "<foo/>"

class DummySend(Service):
    def handle(self):
        resp = self.outgoing.soap.get("testing.dummy").conn.send(self.cid, "<bar/>")
        self.response.payload = "<ok>%s</ok>" % resp.text
~~~

**Comment by dsuch** [comment]
Yes, thanks, let's stick to ```text/xml``` everywhere in SOAP 1.1.

**Comment by dsuch** [comment]
Done and will be released in 2.0.6.
